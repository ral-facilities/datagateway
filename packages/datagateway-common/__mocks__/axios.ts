import axios from 'axios';

const requests = {
  get: vi.fn(() => Promise.resolve({ data: {} })),
  post: vi.fn(() => Promise.resolve({ data: {} })),
  delete: vi.fn(() => Promise.resolve({ data: {} })),
  CancelToken: axios.CancelToken,
  AxiosError: axios.AxiosError,
  isAxiosError: axios.isAxiosError,
};

export default requests;
