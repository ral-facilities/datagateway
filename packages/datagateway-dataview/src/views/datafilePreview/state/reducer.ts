import DATAFILE_PREVIEWER_DEFAULT from '../defaults';
import {
  DecrementDatafilePreviewerZoomLevelType,
  IncrementDatafilePreviewerZoomLevelType,
  ResetDatafilePreviewerZoomLevelType,
  ToggleDatafilePreviewerDetailsPanePayload,
  ToggleDatafilePreviewerDetailsPaneType,
} from './actions';
import { DGDataViewState } from '../../../state/app.types';

/**
 * Defines the shape of the redux state for {@link DatafilePreviewer}
 */
interface DatafilePreviewerState {
  /**
   * Whether the details pane should be shown or not.
   */
  isDetailsPaneShown: boolean;

  /**
   * The current zoom level of the previewer, in percentage. For example,
   * 100% maps to 100, 110% maps to 110, 80% maps to 80, etc.
   */
  zoomLevel: number;
}

function toggleDatafilePreviewerDetailsPane(
  state: DGDataViewState,
  payload: ToggleDatafilePreviewerDetailsPanePayload
): DGDataViewState {
  return {
    ...state,
    datafilePreviewer: {
      ...state.datafilePreviewer,
      isDetailsPaneShown: payload.shouldShow,
    },
  };
}

function incDatafilePreviewerZoomLevel(
  state: DGDataViewState
): DGDataViewState {
  return {
    ...state,
    datafilePreviewer: {
      ...state.datafilePreviewer,
      zoomLevel:
        state.datafilePreviewer.zoomLevel +
        DATAFILE_PREVIEWER_DEFAULT.zoomLevelStep,
    },
  };
}

function decDatafilePreviewerZoomLevel(
  state: DGDataViewState
): DGDataViewState {
  return {
    ...state,
    datafilePreviewer: {
      ...state.datafilePreviewer,
      // zoom level should not go lower than zoomLevelStep
      // otherwise will result in zoom level <= 0.
      zoomLevel: Math.max(
        DATAFILE_PREVIEWER_DEFAULT.minZoomLevel,
        state.datafilePreviewer.zoomLevel -
          DATAFILE_PREVIEWER_DEFAULT.zoomLevelStep
      ),
    },
  };
}

function resetDatafilePreviewerZoomLevel(
  state: DGDataViewState
): DGDataViewState {
  return {
    ...state,
    datafilePreviewer: {
      ...state.datafilePreviewer,
      zoomLevel: DATAFILE_PREVIEWER_DEFAULT.zoomLevel,
    },
  };
}

const datafilePreviewerInitialState: DatafilePreviewerState = {
  isDetailsPaneShown: DATAFILE_PREVIEWER_DEFAULT.showDetailsPane,
  zoomLevel: DATAFILE_PREVIEWER_DEFAULT.zoomLevel,
};

const datafilePreviewerReducer = {
  [ToggleDatafilePreviewerDetailsPaneType]: toggleDatafilePreviewerDetailsPane,
  [IncrementDatafilePreviewerZoomLevelType]: incDatafilePreviewerZoomLevel,
  [DecrementDatafilePreviewerZoomLevelType]: decDatafilePreviewerZoomLevel,
  [ResetDatafilePreviewerZoomLevelType]: resetDatafilePreviewerZoomLevel,
};

export {
  datafilePreviewerInitialState,
  datafilePreviewerReducer,
  toggleDatafilePreviewerDetailsPane,
  incDatafilePreviewerZoomLevel,
  decDatafilePreviewerZoomLevel,
  resetDatafilePreviewerZoomLevel,
};
export type { DatafilePreviewerState };
