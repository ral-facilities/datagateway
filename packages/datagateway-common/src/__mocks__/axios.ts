// TODO: move __mocks__ folder back to package root once facebook/create-react-app#7539 is fixed

import axios from 'axios';

const requests = {
  get: jest.fn(() => Promise.resolve({ data: {} })),
  post: jest.fn(() => Promise.resolve({ data: {} })),
  delete: jest.fn(() => Promise.resolve({ data: {} })),
  CancelToken: axios.CancelToken,
  AxiosError: axios.AxiosError,
};

export default requests;
