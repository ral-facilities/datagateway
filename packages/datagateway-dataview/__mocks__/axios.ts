import axios from 'axios';

const requests = {
  get: vi.fn((path) => {
    if (path === '/datagateway-dataview-settings.json') {
      return Promise.resolve({
        data: {
          facilityName: 'Generic',
          idsUrl: 'ids',
          apiUrl: 'api',
          downloadApiUrl: 'download',
          'ui-strings': '/res/default.json',
        },
      });
    } else {
      return Promise.resolve({
        data: {},
      });
    }
  }),
  post: vi.fn(() => Promise.resolve({ data: {} })),
  delete: vi.fn(() => Promise.resolve({ data: {} })),
  CancelToken: axios.CancelToken,
  isAxiosError: axios.isAxiosError,
};

export default requests;
