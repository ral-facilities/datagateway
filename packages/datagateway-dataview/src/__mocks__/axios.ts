// TODO: move __mocks__ folder back to package root once facebook/create-react-app#7539 is fixed

import axios from 'axios';

const requests = {
  get: jest.fn((path) => {
    if (path === '/datagateway-dataview-settings.json') {
      return Promise.resolve({
        data: {
          facilityName: 'Generic',
          idsUrl: 'ids',
          apiUrl: 'api',
          downloadApiUrl: 'download',
          'ui-strings': '/res/default.json',
        },
      });
    } else {
      return Promise.resolve({
        data: {},
      });
    }
  }),
  post: jest.fn(() => Promise.resolve({ data: {} })),
  delete: jest.fn(() => Promise.resolve({ data: {} })),
  CancelToken: axios.CancelToken,
  isAxiosError: axios.isAxiosError,
};

export default requests;
