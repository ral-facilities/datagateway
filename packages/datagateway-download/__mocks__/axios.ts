import axios from 'axios';

const requests = {
  get: vi.fn((path) => {
    if (path.includes('/topcat/user/cart/')) {
      return Promise.resolve({ data: { cartItems: [] } });
    } else if (path === '/datagateway-download-settings.json') {
      return Promise.resolve({
        data: {
          facilityName: 'LILS',
          apiUrl: 'https://example.com/api',
          downloadApiUrl: 'https://example.com/downloadApi',
          idsUrl: 'https://example.com/ids',
          fileCountMax: 5000,
          totalSizeMax: 1000000000000,
          accessMethods: {
            https: {
              idsUrl: 'https://example.com/ids',
              displayName: 'HTTPS',
              description: 'Example description for HTTPS access method.',
            },
            globus: {
              idsUrl: 'https://example.com/ids',
              displayName: 'Globus',
              description: 'Example description for Globus access method.',
            },
          },
          routes: [
            {
              section: 'Test',
              link: '/download',
              displayName: 'Download',
              order: 0,
            },
          ],
        },
      });
    } else if (path.includes('/topcat/user/downloads')) {
      return Promise.resolve({ data: [] });
    } else {
      return Promise.resolve({ data: {} });
    }
  }),
  post: vi.fn(() => Promise.resolve({ data: {} })),
  put: vi.fn(() => Promise.resolve({ data: {} })),
  CancelToken: axios.CancelToken,
};

export default requests;
