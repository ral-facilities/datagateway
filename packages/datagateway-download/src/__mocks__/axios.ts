// TODO: move __mocks__ folder back to package root once facebook/create-react-app#7539 is fixed

import axios from 'axios';

const requests = {
  get: jest.fn((path) => {
    if (path.includes('/topcat/user/cart/')) {
      return Promise.resolve({ data: { cartItems: [] } });
    } else if (path === '/datagateway-download-settings.json') {
      return Promise.resolve({
        data: {
          facilityName: 'LILS',
          apiUrl: 'https://example.com/api',
          downloadApiUrl: 'https://example.com/downloadApi',
          idsUrl: 'https://example.com/ids',
          fileCountMax: 5000,
          totalSizeMax: 1000000000000,
          accessMethods: {
            https: {
              idsUrl: 'https://example.com/ids',
              displayName: 'HTTPS',
              description: 'Example description for HTTPS access method.',
            },
            globus: {
              idsUrl: 'https://example.com/ids',
              displayName: 'Globus',
              description: 'Example description for Globus access method.',
            },
          },
          routes: [
            {
              section: 'Test',
              link: '/download',
              displayName: 'Download',
              order: 0,
            },
          ],
        },
      });
    } else if (path.includes('/topcat/user/downloads')) {
      return Promise.resolve({ data: [] });
    } else {
      return Promise.resolve({ data: {} });
    }
  }),
  post: jest.fn(() => Promise.resolve({ data: {} })),
  put: jest.fn(() => Promise.resolve({ data: {} })),
  CancelToken: axios.CancelToken,
};

export default requests;
