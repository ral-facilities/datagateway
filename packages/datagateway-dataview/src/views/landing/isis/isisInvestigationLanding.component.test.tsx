import React from 'react';
import { createShallow, createMount } from '@material-ui/core/test-utils';
import ISISInvestigationLanding from './isisInvestigationLanding.component';
import { initialState as dgDataViewInitialState } from '../../../state/reducers/dgdataview.reducer';
import configureStore from 'redux-mock-store';
import { StateType } from '../../../state/app.types';
import {
  dGCommonInitialState,
  fetchInvestigationsRequest,
} from 'datagateway-common';
import { Provider } from 'react-redux';
import thunk from 'redux-thunk';
import { MemoryRouter } from 'react-router';
import axios from 'axios';
import { push } from 'connected-react-router';

describe('ISIS Investigation Landing page', () => {
  let shallow;
  let mount;
  let mockStore;
  let state: StateType;

  const initialData = [
    {
      id: 1,
      title: 'Test 1',
      name: 'Test 1',
      summary: 'foo bar',
      visitId: '1',
      rbNumber: '1',
      doi: 'doi 1',
      size: 1,
      investigationInstruments: [
        {
          id: 1,
          instrument: {
            id: 3,
            name: 'LARMOR',
          },
          investigation: {
            id: 1,
          },
        },
      ],
      studyInvestigations: [
        {
          id: 6,
          investigation: {
            id: 1,
          },
          study: {
            id: 7,
            pid: 'study pid',
            startDate: '2019-06-10',
            endDate: '2019-06-11',
          },
        },
      ],
      startDate: '2019-06-10',
      endDate: '2019-06-11',
      datasets: [
        {
          id: 1,
          name: 'dataset 1',
          doi: 'dataset doi',
        },
      ],
    },
  ];
  const investigationUser = [
    {
      id: 1,
      investigation: {
        id: 1,
      },
      role: 'principal_experimenter',
      user: {
        id: 1,
        name: 'JS',
        fullName: 'John Smith',
      },
    },
    {
      id: 2,
      investigation: {
        id: 1,
      },
      role: 'local_contact',
      user: {
        id: 2,
        name: 'JS',
        fullName: 'Jane Smith',
      },
    },
    {
      id: 3,
      investigation: {
        id: 1,
      },
      role: 'experimenter',
      user: {
        id: 3,
        name: 'JS',
        fullName: 'Jesse Smith',
      },
    },
    {
      id: 4,
      investigation: {
        id: 1,
      },
      role: 'experimenter',
      user: {
        id: 4,
        name: 'JS',
        fullName: '',
      },
    },
  ];
  const sample = [
    {
      id: 1,
      investigation: {
        id: 1,
      },
      name: 'Sample',
    },
  ];
  const publication = [
    {
      id: 1,
      fullReference: 'Journal, Author, Date, DOI',
    },
  ];

  beforeEach(() => {
    shallow = createShallow();
    mount = createMount();

    mockStore = configureStore([thunk]);
    state = JSON.parse(
      JSON.stringify({
        dgdataview: dgDataViewInitialState,
        dgcommon: dGCommonInitialState,
      })
    );
    state.dgcommon.data = initialData;
    state.dgcommon.allIds = [1];

    (axios.get as jest.Mock).mockImplementation(() =>
      Promise.resolve({ data: [] })
    );
    global.Date.now = jest.fn(() => 1);
  });

  afterEach(() => {
    mount.cleanUp();
  });

  it('renders correctly', () => {
    const wrapper = shallow(
      <ISISInvestigationLanding
        store={mockStore(state)}
        instrumentId="4"
        instrumentChildId="5"
        investigationId="1"
        studyHierarchy={false}
      />
    );
    expect(wrapper).toMatchSnapshot();
  });

  it('renders correctly for studyHierarchy', () => {
    const wrapper = shallow(
      <ISISInvestigationLanding
        store={mockStore(state)}
        instrumentId="4"
        instrumentChildId="5"
        investigationId="1"
        studyHierarchy={true}
      />
    );
    expect(wrapper).toMatchSnapshot();
  });

  it('actions dispatched correctly', () => {
    const testStore = mockStore(state);
    const wrapper = mount(
      <Provider store={testStore}>
        <MemoryRouter>
          <ISISInvestigationLanding
            instrumentId="4"
            instrumentChildId="5"
            investigationId="1"
            studyHierarchy={false}
          />
        </MemoryRouter>
      </Provider>
    );

    expect(testStore.getActions()).toHaveLength(1);
    expect(testStore.getActions()[0]).toEqual(fetchInvestigationsRequest(1));

    wrapper.find('#investigation-datasets-tab').first().simulate('click');

    expect(testStore.getActions()).toHaveLength(2);
    expect(testStore.getActions()[1]).toEqual(
      push('/browse/instrument/4/facilityCycle/5/investigation/1/dataset')
    );
  });

  it('actions dispatched correctly in cardView', () => {
    const testStore = mockStore({
      ...state,
      dgcommon: {
        ...state.dgcommon,
        query: { ...state.dgcommon.query, view: 'card' },
      },
    });
    const wrapper = mount(
      <Provider store={testStore}>
        <MemoryRouter>
          <ISISInvestigationLanding
            instrumentId="4"
            instrumentChildId="5"
            investigationId="1"
            studyHierarchy={false}
          />
        </MemoryRouter>
      </Provider>
    );

    expect(testStore.getActions()).toHaveLength(1);
    expect(testStore.getActions()[0]).toEqual(fetchInvestigationsRequest(1));

    wrapper.find('#investigation-datasets-tab').first().simulate('click');

    expect(testStore.getActions()).toHaveLength(2);
    expect(testStore.getActions()[1]).toEqual(
      push(
        '/browse/instrument/4/facilityCycle/5/investigation/1/dataset?view=card'
      )
    );
  });

  it('users displayed correctly', () => {
    const testStore = mockStore(state);
    const wrapper = mount(
      <Provider store={testStore}>
        <MemoryRouter>
          <ISISInvestigationLanding
            instrumentId="4"
            instrumentChildId="5"
            investigationId="1"
            studyHierarchy={false}
          />
        </MemoryRouter>
      </Provider>
    );

    expect(
      wrapper.find('[aria-label="landing-investigation-users-label"]')
    ).toHaveLength(0);
    expect(
      wrapper.find('[aria-label="landing-investigation-user-0"]')
    ).toHaveLength(0);

    state.dgcommon.data = [
      { ...initialData[0], investigationUsers: investigationUser },
    ];
    wrapper.setProps({ store: mockStore(state) });

    expect(
      wrapper.find('[aria-label="landing-investigation-users-label"]')
    ).toHaveLength(3);
    expect(
      wrapper.find('[aria-label="landing-investigation-user-0"]').first().text()
    ).toEqual('Principal Investigator: John Smith');
    expect(
      wrapper.find('[aria-label="landing-investigation-user-1"]').first().text()
    ).toEqual('Local Contact: Jane Smith');
    expect(
      wrapper.find('[aria-label="landing-investigation-user-2"]').first().text()
    ).toEqual('Experimenter: Jesse Smith');
    expect(
      wrapper.find('[aria-label="landing-investigation-user-3"]')
    ).toHaveLength(0);
  });

  it('publications displayed correctly', () => {
    const testStore = mockStore(state);
    const wrapper = mount(
      <Provider store={testStore}>
        <MemoryRouter>
          <ISISInvestigationLanding
            instrumentId="4"
            instrumentChildId="5"
            investigationId="1"
            studyHierarchy={false}
          />
        </MemoryRouter>
      </Provider>
    );

    expect(
      wrapper.find('[aria-label="landing-investigation-publications-label"]')
    ).toHaveLength(0);
    expect(
      wrapper.find('[aria-label="landing-investigation-publication-0"]')
    ).toHaveLength(0);

    state.dgcommon.data = [{ ...initialData[0], publications: publication }];
    wrapper.setProps({ store: mockStore(state) });

    expect(
      wrapper.find('[aria-label="landing-investigation-publications-label"]')
    ).toHaveLength(3);
    expect(
      wrapper
        .find('[aria-label="landing-investigation-reference-0"]')
        .first()
        .text()
    ).toEqual('Journal, Author, Date, DOI');
  });

  it('samples displayed correctly', () => {
    const testStore = mockStore(state);
    const wrapper = mount(
      <Provider store={testStore}>
        <MemoryRouter>
          <ISISInvestigationLanding
            instrumentId="4"
            instrumentChildId="5"
            investigationId="1"
            studyHierarchy={false}
          />
        </MemoryRouter>
      </Provider>
    );

    expect(
      wrapper.find('[aria-label="landing-investigation-samples-label"]')
    ).toHaveLength(0);
    expect(
      wrapper.find('[aria-label="landing-investigation-sample-0"]')
    ).toHaveLength(0);

    state.dgcommon.data = [{ ...initialData[0], samples: sample }];
    wrapper.setProps({ store: mockStore(state) });

    expect(
      wrapper.find('[aria-label="landing-investigation-samples-label"]')
    ).toHaveLength(3);
    expect(
      wrapper
        .find('[aria-label="landing-investigation-sample-0"]')
        .first()
        .text()
    ).toEqual('Sample');
  });

  it('displays citation correctly when study missing', () => {
    const testStore = mockStore({
      ...state,
      dgcommon: {
        ...state.dgcommon,
        data: [{ ...state.dgcommon.data[0], studyInvestigations: undefined }],
      },
    });
    const wrapper = mount(
      <Provider store={testStore}>
        <MemoryRouter>
          <ISISInvestigationLanding
            instrumentId="4"
            instrumentChildId="5"
            investigationId="1"
            studyHierarchy={false}
          />
        </MemoryRouter>
      </Provider>
    );
    expect(
      wrapper
        .find('[aria-label="landing-investigation-citation"]')
        .first()
        .text()
    ).toEqual('Test 1, doi_constants.publisher.name, https://doi.org/doi 1');
  });

  it('displays citation correctly with one user', () => {
    const testStore = mockStore({
      ...state,
      dgcommon: {
        ...state.dgcommon,
        data: [
          {
            ...state.dgcommon.data[0],
            investigationUsers: [investigationUser[0]],
          },
        ],
      },
    });
    const wrapper = mount(
      <Provider store={testStore}>
        <MemoryRouter>
          <ISISInvestigationLanding
            instrumentId="4"
            instrumentChildId="5"
            investigationId="1"
            studyHierarchy={false}
          />
        </MemoryRouter>
      </Provider>
    );
    expect(
      wrapper
        .find('[aria-label="landing-investigation-citation"]')
        .first()
        .text()
    ).toEqual(
      'John Smith; 2019: Test 1, doi_constants.publisher.name, https://doi.org/doi 1'
    );
  });

  it('displays citation correctly with multiple users', () => {
    const testStore = mockStore({
      ...state,
      dgcommon: {
        ...state.dgcommon,
        data: [
          { ...state.dgcommon.data[0], investigationUsers: investigationUser },
        ],
      },
    });
    const wrapper = mount(
      <Provider store={testStore}>
        <MemoryRouter>
          <ISISInvestigationLanding
            instrumentId="4"
            instrumentChildId="5"
            investigationId="1"
            studyHierarchy={false}
          />
        </MemoryRouter>
      </Provider>
    );
    expect(
      wrapper
        .find('[aria-label="landing-investigation-citation"]')
        .first()
        .text()
    ).toEqual(
      'John Smith et al; 2019: Test 1, doi_constants.publisher.name, https://doi.org/doi 1'
    );
  });
});
