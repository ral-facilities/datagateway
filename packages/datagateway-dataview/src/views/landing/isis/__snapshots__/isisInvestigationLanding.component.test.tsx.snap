// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ISIS Investigation Landing page renders correctly 1`] = `
<ContextProvider
  value={null}
>
  <LandingPage
    data={
      Array [
        Object {
          "datasets": Array [
            Object {
              "doi": "dataset doi",
              "id": 1,
              "name": "dataset 1",
            },
          ],
          "doi": "doi 1",
          "endDate": "2019-06-11",
          "id": 1,
          "investigationInstruments": Array [
            Object {
              "id": 1,
              "instrument": Object {
                "id": 3,
                "name": "LARMOR",
              },
              "investigation": Object {
                "id": 1,
              },
            },
          ],
          "name": "Test 1",
          "rbNumber": "1",
          "size": 1,
          "startDate": "2019-06-10",
          "studyInvestigations": Array [
            Object {
              "id": 6,
              "investigation": Object {
                "id": 1,
              },
              "study": Object {
                "endDate": "2019-06-11",
                "id": 7,
                "pid": "study pid",
                "startDate": "2019-06-10",
              },
            },
          ],
          "summary": "foo bar",
          "title": "Test 1",
          "visitId": "1",
        },
      ]
    }
    fetchFacilityCycleData={[Function]}
    fetchStudyData={[Function]}
    instrumentChildId="5"
    instrumentId="4"
    investigationId="1"
    store={
      Object {
        "clearActions": [Function],
        "dispatch": [Function],
        "getActions": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
      }
    }
    studyHierarchy={false}
    view={null}
    viewDatasets={[Function]}
  />
</ContextProvider>
`;

exports[`ISIS Investigation Landing page renders correctly for studyHierarchy 1`] = `
<ContextProvider
  value={null}
>
  <LandingPage
    data={
      Array [
        Object {
          "datasets": Array [
            Object {
              "doi": "dataset doi",
              "id": 1,
              "name": "dataset 1",
            },
          ],
          "doi": "doi 1",
          "endDate": "2019-06-11",
          "id": 1,
          "investigationInstruments": Array [
            Object {
              "id": 1,
              "instrument": Object {
                "id": 3,
                "name": "LARMOR",
              },
              "investigation": Object {
                "id": 1,
              },
            },
          ],
          "name": "Test 1",
          "rbNumber": "1",
          "size": 1,
          "startDate": "2019-06-10",
          "studyInvestigations": Array [
            Object {
              "id": 6,
              "investigation": Object {
                "id": 1,
              },
              "study": Object {
                "endDate": "2019-06-11",
                "id": 7,
                "pid": "study pid",
                "startDate": "2019-06-10",
              },
            },
          ],
          "summary": "foo bar",
          "title": "Test 1",
          "visitId": "1",
        },
      ]
    }
    fetchFacilityCycleData={[Function]}
    fetchStudyData={[Function]}
    instrumentChildId="5"
    instrumentId="4"
    investigationId="1"
    store={
      Object {
        "clearActions": [Function],
        "dispatch": [Function],
        "getActions": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
      }
    }
    studyHierarchy={true}
    view={null}
    viewDatasets={[Function]}
  />
</ContextProvider>
`;
